### ISSUE TITLE
Enable Parquet extension tests for vector size = 2

### ISSUE THREAD

**Issue by Mytherin** [duckdb/duckdb#1539]
There are several issues when running the parquet extension with vector size = 2, which is not tested in the CI yet. These are likely to be actual problems in the code that are not run into with our regular test set but are exposed with smaller vector sizes, hence they are important to fix.

**Comment by hannesmuehleisen** [comment]
Most issues seem to come from using the parquet *writer* with small vector sizes. Otherwise tests look good.

**Comment by Mytherin** [comment]
Triggered by these tests failing locally again I have revisited it [in my branch here](https://github.com/Mytherin/duckdb/tree/parquetvsize2). Indeed it seems that most issues are from the Parquet writer. The reason this fails is that the Parquet writer writes the validity mask in a byte-aligned manner, which fails when the vector size is not byte-aligned (e.g. for vsize=2 it will write 2 bits, then skip the next 6 bytes, write 2 bits, etc). I have disabled the Parquet writer on small vector sizes which seems to fix most of the tests.

However, [these tests have uncovered a problem with reading lists](https://github.com/Mytherin/duckdb/runs/2842875640?check_suite_focus=true). It seems the Parquet reader does not correctly handle child nodes of LIST columns that are bigger than the vector size, which leads to test failures on both vsize=2 or vsize=8.
